Yafc/Workspace/ProductionTable/ProductionLinkSummaryScreen.cs

```diff
@@ -39,8 +39,92 @@ private void BuildScrollArea(ImGui gui) {
             gui.BuildText((amount > 0 ? "Overproduction: " : "Overconsumption: ") + DataUtils.FormatAmount(MathF.Abs(amount), link.flowUnitOfMeasure),
                 new TextBlockDisplayStyle(Font.subheader, Color: SchemeColor.Error));
         }
+        ShowRelatedLinks(gui);
     }
 
+    private void ShowRelatedLinks(ImGui gui) {
+        ModelObject o = link;
+        List<ModelObject> parents = [];
+        while (o.ownerObject is not ProjectPage) {
+            var t = o.ownerObject;
+            if (t is null) {
+                break;
+            }
+            o = t;
+            if (t == o.ownerObject) continue;
+            parents.Add(t);
+        }
+        if (o is ProductionTable page) {
+            Dictionary<ProductionLink, List<(RecipeRow row, float flow)>> childLinks = [], parentLinks = [], otherLinks = [];
+            List<(RecipeRow row, float flow)> unlinked = [], table;
+            foreach (var (r, relationLinks) in
+                GetAllRecipes(page, link.owner)
+                .Select(e => (e, IsParrent(e, parents) ? parentLinks : otherLinks))
+                .Concat(GetAllRecipes(link.owner).Select(e => (e, childLinks)))) {
+                if (link.capturedRecipes.Any(e => e == r)
+                    || (!r.Ingredients.Any(e => e.Goods == link.goods)
+                    && !r.Products.Any(e => e.Goods == link.goods)
+                    && !(r.fuel is not null && r.fuel == link.goods))) {
+                    continue;
+                }
```

Comment:
Can you explain (in a comment maybe) what you are doing here? I cannot make much of it with all the (ultra) compact checks and stuff... (I am not too used to this (C#) syntax)

Like "Iterate over XYZ" and "Check if recipe ..." (at least I guess `r` is for recipe)

Reply:
I think a white line just in front of the `if` statement could help as well. (It took me 3 reads to notice that is separate of the `foreach` due to the unfortunate indentation of the the multi-line `foreach` and the indentation of the block containing the `if` statement...

Same with the `continue` statement with the same indentation of the multi-line `if` conditions...

This all makes it a big code blob, while they are basically 3 separate blocks/expressions.

Reply:
The first 3 lines are part of the for, and retrieve all recipes, first all recipes not nested on the current line, then all nested recipes (childeren).
The next 4 lines are an if filter to ignore the current link (first line) and unrelated links (the next 3 lines).

I'll refactor that into some functions.